frontend/readme.md

```diff
@@ -0,0 +1,76 @@
+# FrontEnd For Streamlit
+
+This Folder contains the frontend code for the OR Assistant using Streamlit. Follow the instructions below to set up the environment, run the application, and perform testing using a mock API.
+
+## Preparing the Environment Variables
+
+To configure the application, you'll need to set up the following environment variables in your `.env` file:
+
+```
+CHAT_ENDPOINT=<your-chat-endpoint> Defaults to http://localhost:8000/chatApp
+```
+
+## For Feedback Evaluation (Optional)
+
+To collect feedback, you need to set up a Google Sheet and configure the necessary environment variables:
+
+1. **Create a Google Sheet**:
+   - Give edit access to the sheet to the service account email.
+   - Add the sheet ID in the environment variable.
+
+    ```plaintext
+    FEEDBACK_SHEET_ID=your-google-sheet-id
+    ```
+
+2. **Using Another Worksheet Inside the Google Sheet**:
+   - If you are using another worksheet inside the Google Sheet, enter the GID as well.
+
+    ```plaintext
+    FEEDBACK_SHEET_GID=your-gid
+    ```
+
+3. **Set Up Google Cloud Console**:
+   - Open Google Cloud Console, enable the Sheets API, and create a service account.
+   - Give access to this service account email.
+   - Export credentials as JSON and add its path in the environment variable.
+
+    ```plaintext
+    GOOGLE_CREDENTIALS_JSON=<path-to-your-google-credentials-json>
+    ```
+4. **Set the Current Version for Feedback Evaluation:**
+   - Add the current version of the feedback evaluation to the environment variables.
+    
+    ```plaintext
+    RAG_VERSION=<current-version>
```

Comment:
Can you automate this? This should be tied to commit number/ some sort of identifier that can be used to trace back

Reply:
I think it would be much better if we integrate this functionality directly into our backend. This way, the RAG_VERSION can be retrieved from the backend API itself.

So, here's what I am thinking
- Use Git tags to check for any new tags when a PR is merged.
- If the tag matches a pattern like bump-version, it will automatically update the RAG_VERSION (e.g., from v0.0.1 to v0.0.2).
- This version update will happen through a commit, allowing us to track version history effectively.

If you have any suggestions do let me know.

Reply:
Sounds good. If you are doing tags then also add a CHANGELOG.md file.

Reply:
> Sounds good. If you are doing tags then also add a CHANGELOG.md file.

Yes, adding a CHANGELOG.md file will definitely make the versioning history more accessible and easier to read. I will work on implementing this.

Reply:
Ok, let's target this in future PRs.